include/deal.II/lac/vector.templates.h

```diff
@@ -304,7 +304,7 @@ void Vector<Number>::reinit (const size_type n,
     }
 
   if (omit_zeroing_entries == false)
-    *this = static_cast<Number>(0);
+    *this = Number(0);
```

Comment:
I suspect you can. We are using both styles numerous times in this file. Let me know which way you want it.